frame/composable-traits/src/oracle.rs

```diff
@@ -39,18 +48,32 @@ pub trait Oracle {
 	/// price (Vaulted base stock_dilution_rate) = price base * stock_dilution_rate
 	/// ```
 	fn get_price(
-		asset: Self::AssetId,
+		asset_id: Self::AssetId,
 		amount: Self::Balance,
 	) -> Result<Price<Self::Balance, Self::Timestamp>, DispatchError>;
 
-	/// Check whether the provided `asset` is supported (a.k.a. a price can be computed) by the
+	/// Check whether the provided `asset_id` is supported (a.k.a. a price can be computed) by the
 	/// oracle.
 	fn is_supported(asset: Self::AssetId) -> Result<bool, DispatchError> {
-		Self::get_price(asset, asset.unit()).map(|_| true)
+		let exponent = Self::LocalAssets::decimals(asset)?;
+		let unit: Self::Balance = 10_u64.pow(exponent).into();
+		Self::get_price(asset, unit).map(|_| true)
 	}
 
 	fn get_twap(
 		of: Self::AssetId,
 		weighting: Vec<Self::Balance>,
 	) -> Result<Self::Balance, DispatchError>;
+
+	/// Up to oracle how it decides ratio.
+	/// If there is no direct trading pair, can estimate via common pair (to which all currencies
+	/// are normalized). General formula
+	/// ```rust
+	/// let base_in_common = 1000.0;
+	/// let quote_in_common = 100.0;
+	/// let ratio = base_in_common / quote_in_common; // 10.0
+	/// let base_amount = 3.0;
+	/// let needed_base_for_quote = base_amount * ratio; // 300.0
+	/// ```
+	fn get_ratio(pair: CurrencyPair<Self::AssetId>) -> Result<FixedU128, DispatchError>;
```

Comment:
should we use typename `Ratio` aliased to `FixedU128` ?

Reply:
fixed in next PR where use this fucntion